diff --git a/src/dbse.c b/src/dbse.c
index fef6a46..4a53a20 100644
--- a/src/dbse.c
+++ b/src/dbse.c
@@ -110,7 +110,7 @@ void dbse_global_init(struct context **cntlist)
                 ,_("SQLite3 Database filename %s")
                 ,cntlist[0]->conf.database_dbname);
 
-            int thread_safe = sqlite3_threadsafe();
+            int thread_safe = 0;
             if (thread_safe > 0) {
                 MOTION_LOG(NTC, TYPE_DB, NO_ERRNO, _("SQLite3 is threadsafe"));
                 MOTION_LOG(NTC, TYPE_DB, NO_ERRNO, _("SQLite3 serialized %s")
