diff --git a/src/util.c b/src/util.c
index 90e81d0..69b4b86 100644
--- a/src/util.c
+++ b/src/util.c
@@ -400,6 +400,14 @@ static void mystrftime_long (const struct context *cnt, int width, const char *w
         sprintf(out, "%*s", width,  cnt->eventid);
         return;
     }
+    if (SPECIFIERWORD("eventtime")) {
+        sprintf(out, "%*lu", width, cnt->eventtime);
+        return;
+    }
+    if (SPECIFIERWORD("timet")) {
+        sprintf(out, "%*lu", width, time(NULL));
+        return;
+    }
 
     // Not a valid modifier keyword. Log the error and ignore.
     MOTION_LOG(ERR, TYPE_ALL, NO_ERRNO,
